src/Microsoft.ApplicationInsights/Extensibility/Implementation/Endpoints/EndpointProvider.cs

```diff
@@ -0,0 +1,180 @@
+        /// <returns>Returns a <see cref="Uri" /> for the requested endpoint.</returns>
+        public Uri GetEndpoint(EndpointName endpointName)
+        {
+            // 1. check for explicit endpoint (location is ignored)
+            // 2. check for endpoint suffix (location is optional)
+            // 3. use classic endpoint (location is ignored)
+
+            var endpointMeta = EndpointMetaAttribute.GetAttribute(endpointName);
+
+            try
+            {
+                if (this.connectionStringParsed.TryGetValue(endpointMeta.ExplicitName, out string explicitEndpoint))
+                {
+                    try
+                    {
+                        return new Uri(explicitEndpoint);
+                    }
+                    catch (UriFormatException ex)
+                    {
+                        throw new ConnectionStringInvalidEndpointException($"The connection string endpoint is invalid. EndpointName: {endpointName} EndpointProperty: {endpointMeta.ExplicitName}", ex);
+                    }
+                }
+                else if (this.connectionStringParsed.TryGetValue("EndpointSuffix", out string endpointSuffix))
+                {
+                    try
+                    {
+                        return BuildUri(
+                            prefix: endpointMeta.EndpointPrefix,
+                            suffix: endpointSuffix,
+                            location: this.GetLocation());
+                    }
+                    catch (UriFormatException ex)
+                    {
+                        throw new ConnectionStringInvalidEndpointException($"The connection string endpoint is invalid. EndpointName: {endpointName} Either EndpointSuffix or Location.", ex);
+                    }
+                }
+                else
+                {
+                    return new Uri(endpointMeta.Default);
+                }
+            }
+            catch (Exception ex)
+            {
+                // TODO: LOG TO ETW ERROR: exception trying to get endpointName. Log Inner Exception.
+                throw; // Re-throw original exception
+            }
+        }
+
+        /// <summary>
+        /// Will evaluate connection string and return the requested instrumentation key.
+        /// </summary>
+        /// <returns>Returns the instrumentation key from the connection string.</returns>
+        public string GetInstrumentationKey()
+        {
+            if (this.connectionStringParsed.TryGetValue("InstrumentationKey", out string value))
+            {
+                return value;
+            }
+            else
+            {
+                throw new ConnectionStringMissingInstrumentationKeyException();
+            }
+        }
+
+        /// <summary>
+        /// Parse a string connection string and return a Dictionary.
+        /// </summary>
+        /// <remarks>Example: "key1=value1;key2=value2;key3=value3".</remarks>
```

Comment:
Do have ABNF definition for the connection string? Would like to see it in the doc string.

Reply:
A general question around the connection string format across Azure - it will be great to follow the connection string convention - for example, if other Azure services allow `OWS` (Optional White Space), we should also allow it here.

Reply:
I spoke with Reiley offline. We're not going to block the PR for this, but we will deliver an ABNF definition in the final documentation. I added this to the OneNote.

Reply:
Yep, this is not blocking. Thanks!